src/sentry/api/paginator.py

```diff
@@ -105,7 +106,16 @@ def get_result(self, limit=100, cursor=None):
 
         # TODO(dcramer): this does not yet work correctly for ``is_prev`` when
         # the key is not unique
+        if count_hits:
+            hits = self.count_hits(queryset)
+            max_hits = 1000
```

Comment:
Seems like this (`max_hits`) should be be a method or constructor parameter, if only so it's testable